File: app/src/main/java/org/mifos/mobilewallet/data/fineract/entity/mapper/AccountMapper.java
package org.mifos.mobilewallet.data.fineract.entity.mapper;

import org.mifos.mobilewallet.account.domain.model.Account;
import org.mifos.mobilewallet.data.fineract.entity.accounts.savings.SavingAccount;
import org.mifos.mobilewallet.data.fineract.entity.client.Client;
import org.mifos.mobilewallet.data.fineract.entity.client.ClientAccounts;
import org.mifos.mobilewallet.home.domain.model.ClientDetails;

import java.util.ArrayList;
import java.util.List;

import javax.inject.Inject;
import javax.inject.Singleton;

/**
 * Created by naman on 11/7/17.
 */

@Singleton
public class AccountMapper {

    @Inject
    public AccountMapper() {}

    public List<Account> transform(ClientAccounts clientAccounts) {
        List<Account> accountList = new ArrayList<>();

        if (clientAccounts != null && clientAccounts.getSavingsAccounts() != null
                && clientAccounts.getSavingsAccounts().size() != 0) {

            for (SavingAccount savingAccount : clientAccounts.getSavingsAccounts()) {
                Account account = new Account();
                account.setName(savingAccount.getProductName());
                account.setNumber(savingAccount.getAccountNo());

                accountList.add(account);
            }


        }
        return accountList;
    }

}

File: core/data/src/main/java/org/mifospay/core/data/fineract/entity/mapper/AccountMapper.kt
/*
 * Copyright 2024 Mifos Initiative
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 *
 * See https://github.com/openMF/mobile-wallet/blob/master/LICENSE.md
 */
package org.mifospay.core.data.fineract.entity.mapper

import com.mifospay.core.model.domain.Account
import com.mifospay.core.model.entity.client.ClientAccounts
import javax.inject.Inject

class AccountMapper @Inject constructor(
    private val currencyMapper: CurrencyMapper,
) {

    fun transform(clientAccounts: ClientAccounts?): List<Account> {
        val accountList = mutableListOf<Account>()

        clientAccounts?.savingsAccounts?.forEach { savingAccount ->
            val account = Account(
                name = savingAccount.productName,
                number = savingAccount.accountNo,
                id = savingAccount.id,
                balance = savingAccount.accountBalance,
                currency = currencyMapper.transform(savingAccount.currency),
                productId = savingAccount.productId.toLong(),
            )
            accountList.add(account)
        }
        return accountList
    }
}
